libvast/src/plugin.cpp

```diff
@@ -380,6 +382,55 @@ store_plugin::make_store(system::accountant_actor accountant,
                                                  std::move(path), name());
 }
 
+// -- loader plugin -------------------------------------------------------------
+
+auto stdin_loader_plugin::make_loader(options, const bool_operator*) const
+  -> caf::expected<input_loader> {
+  return []() -> generator<chunk_ptr> {
+    auto in_buf = detail::fdinbuf(STDIN_FILENO, max_chunk_size);
+    in_buf.read_timeout() = read_timeout;
+    auto chunk_str = std::string{};
+    chunk_str.reserve(max_chunk_size);
+    auto current_char = detail::fdinbuf::int_type{};
+    while (current_char != detail::fdinbuf::traits_type::eof()) {
+      auto current_char = in_buf.sbumpc();
+      if (current_char != detail::fdinbuf::traits_type::eof()) {
+        chunk_str += static_cast<char>(current_char);
+      } else if (in_buf.timed_out() && chunk_str.empty()) {
+        co_yield chunk::make_empty();
+        continue;
+      } else if (!in_buf.timed_out()) {
+        if (chunk_str.empty()) {
+          break;
+        }
+      }
+      if (current_char == detail::fdinbuf::traits_type::eof()
+          || chunk_str.size() == max_chunk_size) {
+        auto chunk = chunk::make(std::exchange(chunk_str, {}));
+        co_yield std::move(chunk);
+        if (current_char != detail::fdinbuf::traits_type::eof()) {
+          chunk_str.reserve(max_chunk_size);
+        }
+      }
+    }
+    co_return;
+  };
+}
+
+auto stdin_loader_plugin::make_default_parser(options,
+                                              const bool_operator*) const
+  -> caf::expected<input_parser> {
+  return caf::make_error(ec::unimplemented, "parser currently not implemented");
+}
+
+caf::error stdin_loader_plugin::initialize(data) {
+  return caf::none;
+}
+
+std::string stdin_loader_plugin::name() const {
+  return "stdin_loader_plugin";
```

Comment:
`from std_loader_plugin read json` reads a bit weird to me, so I think this should just be `stdin` instead.

Reply:
I fixed the name in https://github.com/tenzir/vast/pull/2935/commits/ad2821c61f68d9ab92d2cca3bca422f19c580039